ci/ray_ci/automation/docker_tags_lib.py

```diff
@@ -97,6 +100,73 @@ def list_image_tag_suffixes(
     return tag_suffixes
 
 
+def pull_image(image_name: str) -> None:
+    client = docker.from_env()
+    client.images.pull(image_name)
+
+
+def remove_image(image_name: str) -> None:
+    client = docker.from_env()
+    client.images.remove(image_name)
+
+
+def get_ray_commit(image_name: str) -> str:
+    """
+    Get the commit hash of Ray in the image.
+    """
+    client = docker.from_env()
+
+    # Command to grab commit hash from ray image
+    command = "python -u -c 'import ray; print(ray.__commit__)'"
+
+    container = client.containers.run(
+        image=image_name, command=command, remove=True, stdout=True, stderr=True
+    )
+    output = container.decode("utf-8").strip()
+    match = re.search(
+        r"^[a-f0-9]{40}$", output, re.MULTILINE
+    )  # Grab commit hash from output
+
+    if not match:
+        raise Exception(
+            f"Failed to get commit hash from image {image_name}. Output: {output}"
+        )
+    return match.group(0)
+
+
+def check_image_ray_commit(prefix: str, ray_type: str, expected_commit: str) -> None:
+    if ray_type == RayType.RAY:
+        tags = list_image_tags(
+            prefix, ray_type, PYTHON_VERSIONS_RAY, PLATFORMS_RAY, ARCHITECTURES_RAY
+        )
+    elif ray_type == RayType.RAY_ML:
+        tags = list_image_tags(
+            prefix,
+            ray_type,
+            PYTHON_VERSIONS_RAY_ML,
+            PLATFORMS_RAY_ML,
+            ARCHITECTURES_RAY_ML,
+        )
+    tags = [f"rayproject/{ray_type}:{tag}" for tag in tags]
+
+    for i, tag in enumerate(tags):
+        logger.info(f"{i+1}/{len(tags)} Checking commit for tag {tag} ....")
+
+        pull_image(tag)
+        commit = get_ray_commit(tag)
+
+        if commit != expected_commit:
+            print(f"Ray commit mismatch for tag {tag}!")
+            print("Expected:", expected_commit)
+            print("Actual:", commit)
+            sys.exit(42)  # Not retrying the check on Buildkite jobs
+        else:
+            print(f"Commit {commit} match for tag {tag}!")
+
+        if i != 0:  # Only save first pulled image for caching
+            remove_image(tag)
```

Comment:
this probably won't do much if it is like this.

layers will be shared only when the base images are shared. for different cuda versions of images, they do not really share the base.

but I also do not know how to do better..